docs/content/concepts/partitions-schedules-sensors/backfills.mdx

```diff
@@ -33,15 +33,27 @@ width={1737}
 height={335}
 />
 
-#### Single-run backfills
+### Single-run backfills
+
+By default, if you launch a backfill that covers `N` partitions, Dagster will launch `N` separate runs, one for each partition. This works well when code is single-threaded, as it avoids overwhelming it with large amounts of data.
+
+However, if you're using a parallel-processing engine like Spark and Snowflake, you may not need Dagster to help with parallelism. In this case, splitting the backfill into multiple runs can add extra overhead.
 
-By default, if you launch a backfill that covers N partitions, Dagster will launch N separate runs - one for each partition. This works well when your code is single threaded, because it avoids overwhelming it with large amounts of data. However, if you're using a parallel-processing engine like Spark and Snowflake, you often don't need Dagster to help with parallelism, so splitting up the backfill into multiple runs just adds extra overhead.
+Dagster supports backfills that execute as a single run that covers a range of partitions, such as executing a backfill as a single Snowflake query. After the backfill completes, Dagster will track that all the partitions have been filled.
 
-Dagster supports backfills that execute as a single run that covers a range of partitions. For example, this allows you to execute the backfill as a single Snowflake query. After it completes, Dagster will track that all those partitions have been filled.
+To use this approach, code needs to be written in a way that allows it to operate on partition ranges and not just single partitions.
 
-For this to work, your code needs to be written in a way that allows it to operate on partition ranges, instead of just single partitions.
+If your code uses any of the `partition_key`, `asset_partition_key`, `asset_partition_key_for_output`, or `asset_partition_key_for_input` context methods or properties, you'll need to update the code to use methods or properties like the following instead:
 
-If your code uses any of the `partition_key`, `asset_partition_key`, `asset_partition_key_for_output`, or `asset_partition_key_for_input` context methods or properties, you'll need to update it to use methods or properties like `asset_partitions_time_window`, `asset_partition_key_range`, `asset_partition_keys`, `asset_partitions_time_window_for_output`, `asset_partition_key_range_output`, `asset_partitions_time_window_for_input`, or `asset_partition_key_range_for_input` instead.
+- `asset_partitions_time_window`
+- `asset_partition_key_range`
+- `asset_partition_keys`
+- `asset_partitions_time_window_for_output`
+- `asset_partition_key_range_output`
+- `asset_partitions_time_window_for_input`
+- `asset_partition_key_range_for_input`
```

Comment:
We've improved the API docs for all of these methods to include examples, can we update these to by PyObject links, or have a link to the API docs so that readers know to look there for more examples? 

https://docs.dagster.io/_apidocs/execution#dagster.OpExecutionContext.asset_partition_key_for_input

Reply:
Yes!